_posts/2020-12-14-how-to-lock-down-csp-using-swachbuckle.md

```diff
@@ -0,0 +1,223 @@
+---
+layout: post
+title: How to lock down your CSP when using Swashbuckle
+description: In this post we go through the adjustments we need to make to our app to have a strict CSP while using Swashbuckle.
+---
+
+# Introduction
+
+Swashbuckle is a fantastic .NET library that enables developers to generate Swagger- and OpenAPI-compliant documentation for their APIs.
+
+It also bundles [swagger-ui](https://swagger.io/tools/swagger-ui/) which is a tool that allows developers and API consumers to visualise the definition of an API by using the JSON-based OpenAPI document.
+
+Unfortunately, out of the box, swagger-ui is not CSP-friendly as it uses inline `<script>` and `<style>` tags, which are insecure.
```

Comment:
I am a bit of newbie to CSP and why inline styles are problematic for CSP, can you give a link or a bit more context as to why inline styles are a probelm

Reply:
Thanks @mehdikhaleghian!

I reworded that section, can you let me know if it makes things clearer?

Reply:
Thanks again, would you be able to say why they are insecure?

Reply:
sorry my bad I did not see your change fully thanks